### ISSUE TITLE
[Track1] User Agent Can be Override

### ISSUE THREAD

**Issue by dw511214992** [Azure/autorest.typescript#1005]
I do a test to try to override the Use-Agent in Track1 JS SDK, and it succeed. The Following is the details:
Codes:
```
const msRestNodeAuth = require("@azure/ms-rest-nodeauth");
const { ComputeManagementClient } = require("@azure/arm-compute");

msRestNodeAuth.interactiveLogin().then((creds: any) => {
    const client = new ComputeManagementClient(creds, "eece696d-f03b-47a5-9e4c-157d7ad85b20", {
        userAgent: "aaaaaaaaaaaaaaaaaaaaaaaa"
    });
    client.operations.list().then((result: any) => {
        console.log("The result is:");
        console.log(result);
    });
}).catch((err: any) => {
    console.error(err);
});
```
Requests Header captured by [fiddler](https://www.telerik.com/fiddler):

![image](https://user-images.githubusercontent.com/40835867/120439204-d6c9a180-c3b4-11eb-97c3-690301c1c0cb.png)

I think the user-agent should only be extended, instead of overrode. Please check it.

**Comment by ramya-rao-a** [comment]
Thanks for finding this @dw511214992 

@sarangan12, Can you please check with the auto rest crew as to what the behavior is in other languages (both Track 1 & Track 2)?  

Our new SDK guidelines does say that we should only be allowing adding to the default user agent string and not replacing it. 
I guess, we have been using the convenience layer on top of the generated code to ensure we append the user provided value to the default user agent string and then pass it to the generated layer.

@dw511214992, 
Have you found data to suggest that there is a significant of usage data where the user agent string has been replaced?

**Comment by sarangan12** [comment]
@dw511214992 

Thanks a lot for reporting this issue. I will work on fixing the issue and update you.

@ramya-rao-a 

Yes. I will sync up with the autorest crew and get their behavior also and document it here.

**Comment by dw511214992** [comment]
Hi @ramya-rao-a The [telemetry](https://msit.powerbi.com/groups/de995b1a-16c1-4cc7-a49b-bc9f90bc6acd/reports/8a8582d5-8fdf-4014-8cf9-4d183c5db2c8/ReportSection) count JS SDK's request based on User Agent, and logic is:
```
// azure-sdk-for-node
userAgent.Contains("azure-sdk-for-node") or userAgent.Contains("azure-sdk-for-node.js")
// azure-sdk-for-js
userAgent.Contains("ms-rest-js")
```
If it can be override, the data counted in telemetry will be wrong, and many requests coming from JS SDK will not be counted.



<meta name="ProgId" content="Word.Document">
<meta name="Generator" content="Microsoft Word 15">
<meta name="Originator" content="Microsoft Word 15">
<link rel="File-List" href="file:///C:/Users/weidong/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel="themeData" href="file:///C:/Users/weidong/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel="colorSchemeMapping" href="file:///C:/Users/weidong/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>







Node | userAgent.Contains("azure-sdk-for-node") or userAgent.Contains("azure-sdk-for-node.js")
-- | --
JavaScript | userAgent.Contains("ms-rest-js")   // including LRO

**Comment by sarangan12** [comment]
I have synched up with other SDK owners and confirmed that they are infact appending the user provided user-agent value (prepending in case of Track 2). So, I have to do the same for Track 1 & Track 2 SDK Generators. 

For Track 1 SDK Generator, I am planning to use the user-agent value in the following format: `azure-sdk-for-js/<package_name>/<package_version>  ms-rest-js/<package_version><platform_info>`. An example User-Agent String will be `azure-sdk-for-js/@azure/arm-compute/11.2.0 ms-rest-js/10.0.0 (Ubuntu; Linux x86_64; rv:34.0)`.  To this, the user provided value will be appended.

[Note: There is already a default user agent in ms-rest-js. But, that has only ms-rest-js/<version>/<platformspecificdata> which is not sufficient. The above format, that I suggested was based on the Python's recommendation as mentioned [here](https://azure.github.io/azure-sdk/python_implementation.html#telemetry)]

For Track 2, the situation is similar to Track 1 in case of `core-http`. Once, I complete the implementation, I will update this issue.
